docs/dev-environment-setup.md

```diff
@@ -0,0 +1,239 @@
+  * **If using MacOS**: Unfortunately, the pre-installed system Perl cannot be relied upon; certain critical headers are missing. **Use Homebrew to re-install Perl 5 and ensure that is the perl in your `$PATH`**.
+    - `brew update && brew install perl`
+    - it must be installed somewhere with **regular user permissions** such as `/usr/local/Cellar/` - not somewhere like `/Library/` that requires unsafe root permissions to install further Perl modules.
+    - Homebrew will install perl in the appropriate location by default.
+- From the project root, run `make install_perl_tools`. This invokes the `install` target in `schema/Makefile`.
+  * You may again need to troubleshoot individual steps in `make install_perl_tools` on your system; in that case, find the `install` target in `schema/Makefile` and run the steps individually.
+  * This script installs the Perl package manager [cpanm](https://metacpan.org/pod/distribution/App-cpanminus/bin/cpanm) (_aka._ cpanminus) and then invokes it to install Perl dependencies from the `schema/cpanfile`. Namely, these dependencies are:
+    * the Postgres database driver for Perl, [DBD::Pg](https://metacpan.org/pod/DBD::Pg)
+    * the database migration tool, [Sqitch](https://sqitch.org)
+    * the database unit test runner, [pgTAP](https://pgtap.org/)
+  * By default, cpanm runs each package's tests after installation, which can be quite time-consuming. For this reason, the `make install_perl_tools` script uses `cpanm --notest` to skip tests while installing.
+  * If successful, the `post_install_check` target that is run as part of `make install_perl_tools` should output the installed version of Sqitch.
+
+- **Troubleshooting**:
+    - Ensure the version of Postgres installed by asdf in Step 2 was set [as the global version](https://asdf-vm.com/#/core-manage-versions?id=set-current-version) using `asdf global` _before_ installing Sqitch; otherwise, Sqitch may install a separate instance of Postgres.
+    - Sqitch can alternatively be installed on MacOS with `brew tap sqitchers/sqitch && brew install sqitch`.
+    - If you're on **MacOS** and having trouble with Perl libraries ignoring your newly installed (and thus, properly functioning) Perl in favour of an older system Perl, you can use the [`$PERL5LIB` environment variable](http://mvp.kablamo.org/dependencies/perl5lib/) to point it to the desired location. For example, point it to the new Perl you installed with non-root / regular user permissions instead of the root-installed system Perl.
+
+## 5. Install pgTAP
+
+Ensure the postgres service is running with `pg_ctl start`.
+
+```
+cd schema && make install_pgtap
+```
+This clones and builds [pgTAP](https://pgtap.org/), the database unit testing library from source. The cloned repository will be left behind (visible with `git status`), which you can safely remove after the installation is complete.
```

Comment:
We should add the pgtap folder to our .gitignore

Reply:
Although that would prevent the user from noticing and removing it, as it's no longer necessary? Not sure that matters much either way

Reply:
Could the `make install_pgtap` command remove that installation folder at the end of the script automatically?

Reply:
Sounds like a good solution to me

Reply:
Would you want to take that one on, @pbastia ?

Reply:
Sounds good! I created https://youtrack.button.is/issue/GGIRCS-2005 that I can take on